src/sumo/wrapper/_sumo_client.py

```diff
@@ -1,63 +1,42 @@
 import requests
-import logging
 
-from .config import APP_REGISTRATION, TENANT_ID, AUTHORITY_HOST_URI
-from ._auth import Auth
+from .config import APP_REGISTRATION, TENANT_ID
+from ._new_auth import NewAuth
 from ._request_error import AuthenticationError, TransientError, PermanentError
 
-logging.basicConfig(
-    format="%(asctime)s %(levelname)-8s %(message)s",
-    datefmt="%Y-%m-%d %H:%M:%S"
-)
-
-
 class SumoClient:
     def __init__(
         self,
         env,
         access_token=None,
-        logging_level='INFO',
-        write_back=False
+        interactive=False
     ):
         self.env = env
-        self.user_provided_access_token = access_token
-
-        self.logger = logging.getLogger(__name__)
-        self.logger.setLevel(level=logging_level)
+        self.access_token = access_token
 
         if env not in APP_REGISTRATION:
             raise ValueError(f"Invalid environment: {env}")
 
         self.client_id = APP_REGISTRATION[env]['CLIENT_ID']
         self.resource_id = APP_REGISTRATION[env]['RESOURCE_ID']
-        self.authority_uri = AUTHORITY_HOST_URI + '/' + TENANT_ID
-
-        if not self.user_provided_access_token:
-            self.auth = Auth(
-                client_id=self.client_id,
-                resource_id=self.resource_id,
-                authority=self.authority_uri,
-                writeback=write_back
-            )
-
-            self.access_token = self.auth.get_token()
+        
+        self.auth = NewAuth(
+            client_id=self.client_id,
+            resource_id=self.resource_id,
+            tenant_id=TENANT_ID,
+            interactive=interactive
+        )
 
         if env == "localhost":
             self.base_url = f"http://localhost:8084/api/v1"
         else:
             self.base_url = f"https://main-sumo-{env}.radix.equinor.com/api/v1"
 
     def _retrieve_token(self):
-        if self.user_provided_access_token:
-            self.logger.debug("User provided token exists, returning token")
-            return self.user_provided_access_token
-        else:
-            if self.auth.is_token_expired():
-                self.logger.debug("Token is expired, regenerating")
-                self.access_token = self.auth.get_token()
+        if self.access_token:
```

Comment:
Check whether access_token is expired?

Reply:
Atm there is no logic for the user provided access_token, SumoClient assumes that a valid token has been provided. 

If the provided token expires before the SumoClient object dies, should we raise an exception? Or let the user provide a refresh token in the constructor and renew the access_token when it expires?